### PR TITLE
adding dynamic workspace shapefile and raster samples

### PR DESCRIPTION
**Pull request by ldanzinger**
@JamesMBallard can you please review/merge?

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  // Setup file and folder variables
+  QFile f(filePath);
+  QFileInfo fileInfo(f.fileName());
+  QString workspacePath = QUrl(folder).toLocalFile();
+
+  // create a service from the blank MPK
+  QString mapPackagePath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/mpk_blank.mpk";
+  if (m_localMapService)
+    m_localMapService->deleteLater();
+  m_localMapService = new LocalMapService(mapPackagePath, this);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.cpp]: Please add an empty line after the if case logic.

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.cpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  RasterWorkspace* rasterWorkspace = new RasterWorkspace("raster_wkspc", workspacePath, m_localMapService);
+  RasterSublayerSource* source = new RasterSublayerSource(rasterWorkspace->id(), fileInfo.fileName(), m_localMapService);
+
+  // Add the dynamic workspace to the local map service
+  // This must be added before the service is started
+  QList<DynamicWorkspace*> dynamicWorkspaces{rasterWorkspace};
+  m_localMapService->setDynamicWorkspaces(dynamicWorkspaces);
+
+  // Connect to the status change signal of the local map service
+  connect(m_localMapService, &LocalMapService::statusChanged, this, [=]()
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.cpp]: Is it possible to be more specific with the capture or would the list be too long? This ends up capturing everything which looks like a lot of items.

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+      // Add the sub layer to the image layer
+      imageLayer->mapImageSublayers()->append(imageSublayer);
+
+      // Zoom to the layer's extent
+      connect(imageSublayer, &ArcGISMapImageSublayer::doneLoading, this, [this, imageSublayer](Error e)
+      {
+        if (!e.isEmpty())
+          return;
+        Envelope fullExtent = imageSublayer->mapServiceSublayerInfo().extent();
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.cpp]: Add a line after the return.

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  // Setup file and folder variables
+  QFile f(filePath);
+  QFileInfo fileInfo(f.fileName());
+  QString workspacePath = QUrl(folder).toLocalFile();
+
+  // create a service from the blank MPK
+  QString mapPackagePath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/mpk_blank.mpk";
+  if (m_localMapService)
+    m_localMapService->deleteLater();
+  m_localMapService = new LocalMapService(mapPackagePath, this);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp]: Please add empty line after the if case logic `m_localMapService->deleteLater();`

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        // apply symbology depending on geometry
+        Symbol* symbol = nullptr;
+        GeometryType geomType = imageSublayer->mapServiceSublayerInfo().geometryType();
+        if (geomType == GeometryType::Point || geomType == GeometryType::Multipoint)
+          symbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, QColor("red"), 14.0, imageSublayer);
+        else if (geomType == GeometryType::Polygon)
+          symbol = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor("red"), imageSublayer);
+        else
+          symbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor("red"), 3.0, imageSublayer);
+        SimpleRenderer* simpleRenderer = new SimpleRenderer(symbol, imageSublayer);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp]: Empty line after the last else case logic.

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+  // create a service from the blank MPK
+  QString mapPackagePath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/mpk_blank.mpk";
+  if (m_localMapService)
+    m_localMapService->deleteLater();
+  m_localMapService = new LocalMapService(mapPackagePath, this);
+
+  // Create a shapefile workspace
+  ShapefileWorkspace* shapefileWorkspace = new ShapefileWorkspace("shp_wkspc", workspacePath, m_localMapService);
+  TableSublayerSource * source = new TableSublayerSource (shapefileWorkspace->id(), fileInfo.fileName(), m_localMapService);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp]: nitpick, but `TableSublayerSource(` (get rid of the space)

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+  // Add the dynamic workspace to the local map service
+  // This must be added before the service is started
+  QList<DynamicWorkspace*> dynamicWorkspaces{shapefileWorkspace};
+  m_localMapService->setDynamicWorkspaces(dynamicWorkspaces);
+
+  // Connect to the status change signal of the local map service
+  connect(m_localMapService, &LocalMapService::statusChanged, this, [=]()
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp]: same capture question.

**Comment by ldanzinger** [comment]
@JamesMBallard thanks, updated

**Pull request merged**